### DISCUSSION TITLE
🚀 Feature Request: Apply D1 migrations by database_id and script location

### DISCUSSION THREAD

**Discussion by bruceharrison1984** [cloudflare/workers-sdk#13537] [Feature Requests]
### Describe the solution

When creating our CI/CD, we'd like to split up the build/deploys phases so we can capture the artifacts first, then deploy them in subsequent steps. D1 causes issues here, because the CLI relies exclusively on the `toml` file in order to locate it's ID as well as migrations. 

This requires some additional packaging to move the `toml` file with the migrations scripts. Other than the values contained within it (which we already have) the `toml` file doesn't seem necessary in this scenario. It also requires multiple `toml` files that have the settings for each environment, which seems unnecessary.

It would be nice if `migrations_dir` and `database_id` were available to the CLI in order to perform migrations sans a `toml` file. Ideally, the command would be similar to the following:

```sh
> wrangler d1 migrations apply --database-id <GUID> --migrations-dir ./migrations
```

**Comment by rozenmd** [comment]
That's a fair point, we should handle this

**Comment by dario-piotrowicz** [comment]
As discussed with @rozenmd I'm happy to have a look into this whenever it'll be possible (although we are not too sure when)
